core/client/app/services/time-zone.js

```diff
@@ -0,0 +1,40 @@
+import Ember from 'ember';
+
+const {
+    Service,
+    computed,
+    inject: {service},
+    RSVP: {Promise}
+} = Ember;
+
+export default Service.extend({
+    store: service(),
+    _settings: null,
+
+    _parseTimezones(settings) {
+        let activeTimezone = settings.get('activeTimezone');
+
+        try {
+            let [ , offset ] = activeTimezone;
+            return offset || {};
+        } catch (e) {
+            return {};
+        }
+    },
+
+    offset: computed('_settings', function () {
+        return new Promise((resolve, reject) => {
+            if (this.get('_settings')) { // So we don't query the backend every single time
```

Comment:
By default the result of a CP's function will be cached so subsequent calls to `timeZone.get('offset')` will always return the same promise meaning that the API query will only be performed on the first request. Combining that with [queryRecord](http://emberjs.com/blog/2015/06/18/ember-data-1-13-released.html#toc_reorganized-find-methods) could simplify this CP to:

```javascript
offset: computed(function () {
    let store = this.get('store');

    return store.queryRecord('setting', {type: 'blog'}).then((settings) => {
        return this._parseTimezones(settings);
    });
})
```